### PR TITLE
Remove `dmapped` from docs

### PR DESCRIPTION
**Pull request by jeremiah-corrado**
Replace various uses of `dmapped` throughout the docs with distribution factory functions due to `dmapped` being unstable.

Also adds documentation for the `createDomain` methods on `blockDist`, `cyclicDist`, and `stencilDist` that were missing documentation.

- [x] paratest
- [x] inspected docs

### CODE REVIEW

File: modules/dists/BlockDist.chpl [hunk 1]
```diff
@@ -324,6 +326,15 @@ The helper methods on ``blockDist`` have the following signatures:
     .. Warning::
       ``blockDist.createArray`` with an ``initExpr`` formal is unstable and may change in a future release
 
+  .. function:: proc blockDist.createDomain(dom: domain(?))
+
+    Create a block-distributed domain over an existing ``blockDist``.
```
Comment (mppf) [inline modules/dists/BlockDist.chpl:331]: Might want to say, "by copying the index space from the passed domain"

File: modules/dists/CyclicDist.chpl [hunk 2]
```diff
@@ -226,6 +228,14 @@ The helper methods on ``Cyclic`` have the following signatures:
     * an array of compatible size and type — the array will be assigned into
       the distributed array
 
+  .. function:: proc cyclicDist.createDomain(dom: domain(?))
+
+    Create a cyclic-distributed domain over an existing ``cyclicDist``.
```
Comment (mppf) [inline modules/dists/CyclicDist.chpl:233]: same here

File: modules/dists/StencilDist.chpl [hunk 3]
```diff
@@ -262,6 +264,15 @@ config param disableStencilLazyRAD = defaultDisableLazyRADOpt;
     * an array of compatible size and type — the array will be assigned into
       the distributed array
 
+  .. function:: proc stencilDist.createDomain(dom: domain(?))
+
+    Create a stencil-distributed domain over an existing ``blockDist``.
```
Comment (mppf) [inline modules/dists/StencilDist.chpl:269]: and here

File: modules/packages/LinearAlgebra.chpl [hunk 4]
```diff
@@ -3480,11 +3480,11 @@ module Sparse {
 
 
   @chpldoc.nodoc
-  /* Returns ``true`` if the array is dmapped to ``CS`` layout. */
+  /* Returns ``true`` if the array is mapped to the ``CS`` layout. */
```
Comment (mppf) [inline modules/packages/LinearAlgebra.chpl:3483]: I thought we would say "if the array is distributed with the ``CS`` layout" but in this case I think it's better to say "if the array uses the ``CS`` layout".

File: modules/packages/LinearAlgebra.chpl [hunk 5]
```diff
@@ -3480,11 +3480,11 @@ module Sparse {
...
   proc isCSArr(A: []) param { return isCSType(A.domain.distribution.type); }
 
   @chpldoc.nodoc
-  /* Returns ``true`` if the domain is dmapped to ``CS`` layout. */
+  /* Returns ``true`` if the domain is mappd to the ``CS`` layout. */
```
Comment (mppf) [inline modules/packages/LinearAlgebra.chpl:3487]: Ditto, also spelling error here

File: test/release/examples/primers/distributions.chpl [hunk 6]
```diff
@@ -161,9 +177,26 @@ for (L, ML) in zip(BA2.targetLocales(), MyLocales) do
...
+/*
+As with ``blockDist``, the above could also be shortened to one of the
+following expressions for convenience:
+
+.. code-block:: chapel
+
+	const CyclicSpace = cyclicDist.createDomain(Space);  
+	const CA: [CyclicSpace] int;
+
+or:c
```
Comment (mppf) [inline test/release/examples/primers/distributions.chpl:192]: Extra `:c` here

File: test/release/examples/primers/forallLoops.chpl [hunk 7]
```diff
@@ -150,14 +150,14 @@ Its parallel iterator will determine how this loop is parallelized.
...
+as well as arrays over such domains, provide both serial and parallel
 iterators. So do domains distributed over standard distributions,
-such as blockDist and cyclicDist (:ref:`primers-distributions`), and
-arrays over such domains. The parallel iterators provided
-by standard distributions place each loop iteration on the
-locale where the corresponding index or array element is placed.
+such as blockDist and cyclicDist, and arrays over such domains. The
+parallel iterators provided by standard distributions place each loop
+iteration on the locale where the corresponding index or array element
+is placed.
```
Comment (mppf) [inline test/release/examples/primers/forallLoops.chpl:160]: Yeah, but we are aiming for our terminology to consider something like DefaultRectangular or LayoutCS as degenerate "distributions". As a result, we might say that these are standard "distributions" (in that they are in the standard library). Seems like a bit of a terminology puzzle here but I'd probably resolve it by calling Block and Cyclic here "multi-locale distributions":

> So do domains distributed over standard multi-locale distributions,
> such as blockDist and cyclicDist, and arrays over such domains. The
> parallel iterators provided by standard multi-locale distributions place each loop
> iteration on the locale where the corresponding index or array element
> is placed.

**Review by mppf — APPROVED** [review]

File: modules/dists/CyclicDist.chpl [hunk 8]
```diff
@@ -97,7 +97,7 @@ to the ID of the locale to which it is mapped.
     use CyclicDist;
 
     const Dist = new cyclicDist(startIdx=(1,1));
-    const D = {1..8, 1..8} dmapped Dist;
+    const D = cyclicDist.createDomain({1..8, 1..8});
```
Comment (Guillaume-Helbecque) [inline modules/dists/CyclicDist.chpl:100]: I've just come across this typo in the doc, should be `const D = Dist.createDomain({1..8, 1..8})`, right?

Comment (jeremiah-corrado) [reply modules/dists/CyclicDist.chpl:100]: Good catch, thanks! I'll fix it.

**Pull request merged**